res/controllers/midi-components-0.0.js

```diff
@@ -651,7 +651,13 @@
             // Unset isShifted for each ComponentContainer recursively
             this.isShifted = false;
         },
+        /**
+         * @param newLayer Layer to apply to this
+         * @param reconnectComponents Whether components should be reconnected or not
+         * @deprecated Use @{Component#setLayer} instead
+         */
         applyLayer: function(newLayer, reconnectComponents) {
+            console.warn("Components.applyLayer is deprecated; use Component.setLayer instead");
```

Comment:
`applyLayer` is used extensively. Do you think its feasible to transition some code to `setLayer` in certain mappings?

Reply:
In this PR? Sure! Every mapping or just one or two?

Reply:
not sure. I'd say every one where you can be sure that the change to `Object.assign` does not cause regressions. Ideally any hardware you have on hand to test/verify that it didn't break anything.

Reply:
Uh. I only have 2 Denon DN-S3700 here, sorry.

Reply:
Do you have an idea aboutwhat are the two most used controllers? I propose to start by them and them ask the community to test.

Reply:
not sure. I'm guessing many pioneer controllers. Apart from that I could test on my NS6II and there are several people in the mixxx team that have a traktor S4MK3 (though that uses its own ComponentsJS implementation IIRC).

Reply:
We could just switch it for all mappings before a beta and see if we receive complaints during the beta phase... Thats what betas are for IMO.

Reply:
So we switch the mappings in a seperate PR ?